torch/_ops.py

```diff
@@ -280,6 +280,58 @@ def key_extractor(tensors, key_mask):
     return key_set
 
 
+@contextlib.contextmanager
```

Comment:
A top level comment explaining what the difference between these functions and the "core" counterparts are would be very useful. Even better to have short docblocks on each one reiterating this.